### PR TITLE
[WIP] Improve Testability and Write Tests

### PR DESCRIPTION
**Pull request by phillipahereza**
Improve testability of the plugin and write tests for the features already implemented

### CODE REVIEW

File: server/client/client.go [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+// Token returns the access token
+func (t *TokenSource) Token() (*oauth2.Token, error) {
+	token := &oauth2.Token{
+		AccessToken: t.AccessToken,
+	}
+	return token, nil
+}
+
+// DigitalOceanService is an interface for interfacing with the DigitalOcean API endpoints
+type DigitalOceanService interface {
```
Comment (AGMETEOR) [inline server/client/client.go]: Maybe it's better to name this Client to be a little bit more consistent with ` DigitalOceanClient` on line 12

**Comment by AGMETEOR** [comment]
@phillipahereza whatever is failing our pipeline is now coming from the webapp component. Trying to investigate that.

File: server/client/client.go (hunk 1 above)
Comment (AGMETEOR) [reply server/client/client.go]: > Maybe it's better to name this Client to be a little bit more consistent with ` DigitalOceanClient` on line 12

Rename to client

Comment (phillipahereza) [reply server/client/client.go]: Won't the name collide with DigitalOceanClient of type struct on line 12

Comment (AGMETEOR) [reply server/client/client.go]: I doubt. As we have no other types named Client

**Comment by phillipahereza** [comment]
Sawa lemme create a PR for clean up tasks, everytime we leave something unclean we add to the list of tasks to do

**Pull request merged**